alvr/graphics/resources/stream.wgsl

```diff
@@ -164,15 +164,30 @@ fn fragment_main(@location(0) uv: vec2f) -> @location(0) vec4f {
 }
 
 fn chroma_key_mask(color: vec3f) -> f32 {
-    let start_max = vec3f(pc.ck_channel0.x, pc.ck_channel1.x, pc.ck_channel2.x);
-    let start_min = vec3f(pc.ck_channel0.y, pc.ck_channel1.y, pc.ck_channel2.y);
-    let end_min = vec3f(pc.ck_channel0.z, pc.ck_channel1.z, pc.ck_channel2.z);
-    let end_max = vec3f(pc.ck_channel0.w, pc.ck_channel1.w, pc.ck_channel2.w);
-
-    let start_mask = smoothstep(start_min, start_max, color);
-    let end_mask = smoothstep(end_min, end_max, color);
-
-    return max(start_mask.x, max(start_mask.y, max(start_mask.z, max(end_mask.x, max(end_mask.y, end_mask.z)))));
+    var start_max = vec3f(pc.ck_channel0.x, pc.ck_channel1.x, pc.ck_channel2.x);
+    var start_min = vec3f(pc.ck_channel0.y, pc.ck_channel1.y, pc.ck_channel2.y);
+    var end_min = vec3f(pc.ck_channel0.z, pc.ck_channel1.z, pc.ck_channel2.z);
+    var end_max = vec3f(pc.ck_channel0.w, pc.ck_channel1.w, pc.ck_channel2.w);
+
+    let sv_value = max(
+        max(
+            smoothstep(start_min.y, start_max.y, color.y),
+            smoothstep(start_min.z, start_max.z, color.z)
+        ),
+        max(
+            smoothstep(end_min.y, end_max.y, color.y),
+            smoothstep(end_min.z, end_max.z, color.z)
+        )
+    );
```

Comment:
you have yet to use swizzling here. it would reduce the amount of smoothsteps to 2, and aide vectorization